### ISSUE TITLE
How do you get a foreground image without background

### ISSUE THREAD

**Issue by gakkitw** [developfeng/MGCAM#3]
How do you use FCN to get a foreground image without background? I am a newcomer to this, can you give me some advice?
I only want to get a pedestrian image without background. What should I do? Thank you.

**Comment by developfeng** [comment]
We get the foreground image with a well-trained  FCN model. We are sorry that we could not share this segmentation model. You can refer to the paper of "Early Hierarchical Contexts Learned by Convolutional Networks for Image Segmentation." and "1000fps human segmentation with deep convolutional neural networks"  to find some details.

**Comment by hjwang-824** [comment]
Hi, I have read that 2 papers and download the baidu segmentation dataset. I trained it with a FCN model, however, the test on the ReID dataset are not good enough. So, have you used some tricks such as preprocess the baidu dataset? Can you share some details? Thank you so much! @developfeng

**Comment by developfeng** [comment]
Hi, there are some differences with above two paper. 1) We re-design and modify the FCN based model motivatied by above two papers. However, the model used in our CVPR paper is not exactly the one in "1000fps human segmentation with deep convolutional neural networks". We are sorry that we can not share the segmentation model now. 2) Similar with the way in the paper "1000fps human segmentation with deep convolutional neural networks", much more extra data is added for better training.

**Comment by hjwang-824** [comment]
Thank you，I will try again.

**Comment by hjwang-824** [comment]
And can you share your extra data if it’s OK to you?

**Comment by developfeng** [comment]
Sorry, we can't share the segmentation data. I think one possible solution is that, you can try to select some released high-quality masks and its RGB images of Market-1501 and MARS to train your model.

**Comment by hjwang-824** [comment]
OK，thank you.

**Comment by bmiftah** [comment]
@Ko8e-24 ... I Do you mean how to crave out the background and keep the foreground( The person ) ? If so , I guess the mask shared in this repository by @developfeng and the original RGB image in  can do that for you.  foreground  =  RGB_Image x Mask_of_RGB_Image.  I tried and here is some sample I got : - 

![original_forehround_mask](https://user-images.githubusercontent.com/28997140/84570254-dceafa80-adbe-11ea-8d87-23ef3a023cd8.png)

@developfeng  Thank you for sharing the mask for Market-1501 and cuhk03. I understand you  didn't start sharing the segementation code you used to get your masks, but I would like to know if you have mask data for DukeMTMC dataset

**Comment by HWNHJJ** [comment]
hello,could you share the mask data for DukeMTMC dataset? Thank you very mach.

**Comment by bmiftah** [comment]
@HWNHJJ    I am also looking for these dataset ... hope @developfeng  could share us sooner .

**Comment by SamSamhuns** [comment]
@developfeng, is the code for the paper "1000fps human segmentation with deep convolutional neural networks" available somewhere?

**Comment by BonBon-Boom** [comment]
> @Ko8e-24 ... I Do you mean how to crave out the background and keep the foreground( The person ) ? If so , I guess the mask shared in this repository by @developfeng and the original RGB image in can do that for you. foreground = RGB_Image x Mask_of_RGB_Image. I tried and here is some sample I got : -
> 
> ![original_forehround_mask](https://user-images.githubusercontent.com/28997140/84570254-dceafa80-adbe-11ea-8d87-23ef3a023cd8.png)
> 
> @developfeng Thank you for sharing the mask for Market-1501 and cuhk03. I understand you didn't start sharing the segementation code you used to get your masks, but I would like to know if you have mask data for DukeMTMC dataset

Thank  you for sharing method to get a foreground, I do as you told ,but why I get this?And ,why your mask is yellow-purple, since author gave us black-white mask?
<img width="47" alt="fg" src="https://user-images.githubusercontent.com/54622777/225021778-8f2556fd-298d-4042-9c97-cc3a2e10403b.png">
